### ISSUE TITLE
Same Model-DB Issue

### ISSUE THREAD

**Issue by ddink** [cmgmyr/laravel-messenger#11]
Hey @cmgmyr I'm sorry to bug you again about the same issue, but it seems that even with the Messagable trait correctly spelled within my User model the Laravel-Messenger's models are not being found.

Like last time, I've got the code inside the MesssagesController's __construct, index, create, and store functions working. But the show and update functions aren't working, because the Thread models aren't being found from the `Thread::findOrFail($id)` statement (it's coming up as a Class Not Found error).

I reconstructed and re-uploaded my app (https://github.com/ddink/echo), because I keep hitting that same dead-end. I've got the model files configured correctly, the database is adequately seeded, and I can even query the database for the contents of their tables. The problem is that those queries--like `(DB::table('threads')->where('id', $id)->first())`--are getting a stdClass object when what I'm trying to get is a Thread object. Because you can't call the Thread model's methods on an object that isn't a Thread object.

Thanks for all your help.

**Comment by cmgmyr** [comment]
@ddink I re-cloned and re-installed your repo and I'm running it fine on my local environment. Are you using the Homestead vagrant box, or something else?

@dawgbone are you using Homestead, or something else? I'm also not seeing this error locally.

**Comment by dawgbone** [comment]
@cmgmyr Basic WAMP setup. Everything works fine except for the show message part, unless, like I said, I delete the $message->user references), then the message shows, but no user info.

I had to make a couple of changes (my user table is userprofiles), so I could potentially be missing another configuration somewhere, but I can't find it.

**Comment by ddink** [comment]
@cmgmyr I'm using the Homestead vagrant box setup, but I am developing in Windows. I don't know if that would make a difference.

I'm also using MySQL Workbench to manage my database.

**Comment by cmgmyr** [comment]
@dawgbone @ddink, have you tried publishing the config file for the package? The `user_model` config defaults to the `User` model, but if you are using a different model, you'll need to publish and update the model reference. If you're using namespaces, you'll need to reference the full namespace (`\Acme\Models\User`)

**Comment by ddink** [comment]
@cmgmyr I'm using the default `User` model that comes with each Laravel installation, and I've published the package's config file.

**Comment by cmgmyr** [comment]
@ddink what's the exception that you are getting, `ModelNotFoundException` again?

**Comment by ddink** [comment]
@cmgmyr yup. A `ModelNotFoundException` whenever I try to visit the `show` route.

**Comment by cmgmyr** [comment]
Ok, I think the thread ID that you are trying to access isn't available. Can you do a `composer update` and look at the adjusted example files? I updated the index view and the MessagesController which will now tell you if a thread hasn't been found. Let me know if that fixes the problem

**Comment by dawgbone** [comment]
@cmgmyr switching my table name from userprofiles to users seems to have fixed the issue I was having. Not sure why that is, the User model was properly referencing my table name.

**Comment by cmgmyr** [comment]
@dawgbone odd, I'll take a look into that to see what might be happening. Do you happen to have your code in a public repo that I could look at? If not, that's ok.

**Comment by dawgbone** [comment]
I can post it tomorrow.

**Comment by cmgmyr** [comment]
@ddink are there messages and participants connected to that thread id?

**Comment by ddink** [comment]
yup:

![echo-messages](https://cloud.githubusercontent.com/assets/2703920/5638786/42e70ee4-95da-11e4-8811-2186731eb744.JPG)


![echo-participants](https://cloud.githubusercontent.com/assets/2703920/5638769/05ba8ea6-95da-11e4-9517-4f88f32bb9c9.JPG)

**Comment by cmgmyr** [comment]
@ddink I'm at a loss there. I can't see any reason why it won't work. Have you tried deploying what you have to a server and see if you had the same issue? The only other thing that I can think of is that you have duplicate models [here](https://github.com/ddink/echo/tree/master/app/models) instead of just having the ones in the package. Can you try deleting those and doing a `composer dump-autoload`? Also maybe try starting with a fresh database and migrate from scratch? I see you also have some duplicate migrations [here](https://github.com/ddink/echo/tree/master/app/database/migrations)

**Comment by ddink** [comment]
Darn. Well, I'll give those a suggestions a try and keep you posted. I had those duplicate models and migrations, because at one point my routes weren't working without the models and errors were coming up in the console when I'd run `php artisan migrate --package=cmgmyr/messenger`. But it could be because I've used that same database with other versions of this app.

**Comment by ddink** [comment]
I got it working! Ok, so here's what I did:

First) I deleted the duplicate models and ran `composer dump-autoload`. Afterwards my app was operating without my duplicate model files.

Second) I deleted the duplication migrations and my database. I created a new database and started with the package's migrations. When I ran `php artisan migrate --package=cmgmyr/messenger`, I encountered an error saying my messages table couldn't be created without `foreign key ('user id')` which `references 'users' ('id')`. 

Third) I deleted my new database and created another fresh one. This time I started by running `php artisan migrate`, which migrated the create_users_table migration. Afterwards, I ran the package's migrations without any errors or issues. 

So basically, the problem all along was my build process: When I first installed laravel-messenger I don't think I had ran a migration for my User model. I think I ran the package's migrations first, and (because the Thread, Message, and Participant models are dependent on the User model) that's probably what caused the issues between the models and the database. I'm going to close this issue. Thanks so much for the help @cmgmyr!
